src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+For more information about static types and `dynamic`, see
+[The Dart type system](/guides/language/sound-dart).
+
+### API conflicts
+
+If an extension member conflicts with
+an interface or with another extension member,
+then you have a few options.
+
+One option is changing how you import the conflicting extension.
+You can use `show` or `hide`, or you can remove the import:
+
+```dart
+// Both libraries define the String extension method toInt().
+// The extensions might have the same name or different names.
+import 'string_apis.dart';
+import 'string_apis_2.dart' hide ParseNumbers2;
+...
+// Use the toInt() defined in 'string_apis.dart'.
+print('42'.toInt());
+```
+
+Another option is applying the extension explicitly, which results in code that looks as if the extension is a wrapper class:
+
+```dart
+// Both libraries define extensions on String that contain toInt(),
+// and the extensions have different names.
+import 'string_apis.dart'; // Contains ParseNumbers extension.
+import 'string_apis_2.dart'; // Contains ParseNumbers2 extension.
+...
+//  print('42'.toInt()); // Doesn't work.
+print(ParseNumbers('42').toInt());
+print(ParseNumbers2('42').toInt());
+```
+
+If both extensions have the same name,
+then you might need to import using a prefix:
+
+```dart
+// Both libraries define extensions named ParseNumbers
+// that contain the extension method toInt(). One ParseNumbers
+// extension (in 'string_apis_3.dart') also defines toNum().
+import 'string_apis.dart';
+import 'string_apis_3.dart' as rad;
+...
+//  print('42'.parseInt()); // Doesn't work.
+
+// Use the ParseNumbers extension from string_apis.dart.
+print(ParseNumbers('42').toInt());
+
+// Use the ParseNumbers extension from string_apis_3.dart.
+print(rad.ParseNumbers('42').toInt());
+
+// Only string_apis_3.dart has toNum().
+print('42'.toNum());
+```
+
+As the example shows,
+you can invoke extension methods implicitly even if you import using a prefix.
+The only time you need to use the prefix is
+to avoid a name conflict when invoking an extension explicitly.
+
+
+## Implementing extension methods
+
+Use the following syntax to create an extension:
+
+```
+extension <extension name> on <type> {
+  (<member definition>)*
+}
+```
+
+For example, here’s how you might implement an extension of the `String` class:
+
+```dart
+// In string_apis.dart:
+extension ParseNumbers on String {
+  int toInt() {
+    return int.parse(this);
+  }
+
+  double toDouble() {
+    return double.parse(this);
+  }
+}
+```
+
+To create a local extension that’s visible only in
+the library where it’s declared,
+either omit the extension name or give it a name
+that starts with an underscore (`_`).
+
+The members of the extension can be methods, getters, setters, operators.
+Extensions can also have static fields and static helper methods.
+
+## Implementing generic extensions
+
+Extensions can have generic type parameters.
+For example, here’s some code that extends the built-in `List<T>` type:
```

Comment:
maybe add "with a getter and two methods"?

Reply:
It's a getter, an operator, and a method, right?